api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -19,92 +19,119 @@
 
```

Comment:
This file now contains the new RewriteDataFiles API The old File is in RewriteDataFilesV1

Reply:
Why do we need `RewriteDataFilesV1` in the new API? I'd say we should only expose the new logic. Let's keep the current implementation under the old interface and old entry point.

It should be safe to simply replace the content of `RewriteDataFiles` as there is no implementation and we delayed the discussion how the API should look like up until now.